### ISSUE TITLE
What could be a reason for "lower losses, comparable performance"?

### ISSUE THREAD

**Issue by Zephyr69** [ultralytics/yolov5#8485]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I have two models trained under exactly the same settings except the model scale, with model A being larger than model B. The dataset is one with small targets (7 ~ 20 pixel wide boxes). Autoanchor is on.

Both train and val losses of model A are significantly smaller than those of model B, but the difference in their metrics ended up negligible. 

What could lead to this?

**Comment by glenn-jocher** [comment]
@Zephyr69 well it's hard to say. We don't maximize the metrics during training, we minimize the losses, which is similar but not exactly the same.

Since we have 3 loss components it's also possible one loss is holding up the metrics (i.e. the weak link in the chain). What does do your results.png images look like?

**Comment by Zephyr69** [comment]
They look like this:
<img width="1019" alt="01" src="https://user-images.githubusercontent.com/63066174/177890568-46434a4d-33d7-49b2-8b17-ec9c5215d496.png">
<img width="773" alt="02" src="https://user-images.githubusercontent.com/63066174/177890574-629aa829-4a8d-48db-9b95-cd22026072d9.png">

**Comment by glenn-jocher** [comment]
@Zephyr69 thanks for the plots, very interesting. I'd say it's not impossible for this to happen but it's pretty rare. The val loss differences are substantial relative to the total loss, so I would expect blue to produce better metrics. There's obviously something we're not understanding here, a weak link in the chain that's preventing metrics improvements despite loss reduction.

Also interesting is that your box and cls loss are overfitting more than your obj loss. Usually it's the opposite, we traditionally have obj loss overfitting more than any other loss, especially when starting from pretrained models.

**Comment by Zephyr69** [comment]
@glenn-jocher Would you say it's likely that something in the NMS is causing the issue? Cuz I found it hard to imagine a situation where the model losses are not very indicative of model performance in this task.

Or what else could it be?

**Comment by glenn-jocher** [comment]
@Zephyr69 it's hard to say, it could indicate a problem somewhere or just something we are not understanding, i.e. constraints somewhere that are not allowing lower losses to produce better metrics.
